CMakeLists.txt

```diff
@@ -5,110 +5,115 @@ include(cmake/GitHelpers.cmake)
 option(CVSLOGGER_ENABLE_SYSD_BY_DEFAULT "Enable systemd sink by default" ON)
 
 include(GenerateExportHeader)
 
 if (NOT SPDLOG_FMT_EXTERNAL OR NOT SPDLOG_FMT_EXTERNAL_HO)
-    message(NOTICE "Set SPDLOG_FMT_EXTERNAL to ON. "
-        "FPSLogger is not compatible with internal fmt in spdlog version 1.8.2 and older.")
-    set(SPDLOG_FMT_EXTERNAL ON CACHE BOOL "")
+  message(NOTICE "Set SPDLOG_FMT_EXTERNAL to ON. "
+    "FPSLogger is not compatible with internal fmt in spdlog version 1.8.2 and older.")
+  set(SPDLOG_FMT_EXTERNAL ON CACHE BOOL "")
 endif()
 
 if(NOT TARGET fmt::fmt)
-    find_package(fmt REQUIRED)
+  find_package(fmt REQUIRED)
 endif()
 
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-add_subdirectory(thirdparty/spdlog)
-
+find_package(spdlog)
 find_package(PkgConfig)
-pkg_check_modules(PC_SYSTEMD QUIET libsystemd)
+pkg_check_modules(systemd QUIET IMPORTED_TARGET libsystemd)
 
 if(CVSLOGGER_OPENCV_IMG)
-    find_package(OpenCV REQUIRED)
+  find_package(OpenCV REQUIRED)
 endif()
 
-file( GLOB SRCS
-        include/cvs/logger/logging.hpp
-        include/cvs/logger/ilogger.hpp
-        include/cvs/logger/loggerfactory.hpp
-        include/cvs/logger/configtypes.hpp
-        include/cvs/logger/tools/fpslogger.hpp
-
-        src/default/defaultfactory.hpp
-
-        src/default/defaultfactory.cpp
-        src/tools/fpslogger.cpp
-        src/configtypes.cpp
-        src/loggerfactory.cpp
-        src/ilogger.cpp
-    )
 
 if(CVSLOGGER_SHARED)
-    add_library(${PROJECT_NAME} SHARED)
+  add_library(${PROJECT_NAME} SHARED)
 else()
-    add_library(${PROJECT_NAME} STATIC)
+  add_library(${PROJECT_NAME} STATIC)
 endif()
 
-target_sources(${PROJECT_NAME}
-    PRIVATE
-        ${SRCS}
-    )
+set_target_properties(${PROJECT_NAME} PROPERTIES
+  VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+  )
 
 generate_export_header(${PROJECT_NAME}
-    EXPORT_FILE_NAME
-        ${CMAKE_CURRENT_BINARY_DIR}/export/cvs/logger/${PROJECT_NAME}_export.hpp
-    )
+  EXPORT_FILE_NAME
+    ${CMAKE_CURRENT_BINARY_DIR}/export/cvs/logger/${PROJECT_NAME_LOWER}export.hpp)
+
+target_sources(${PROJECT_NAME}
+  PRIVATE
+    include/cvs/logger/logging.hpp
+    include/cvs/logger/ilogger.hpp
+    include/cvs/logger/loggerfactory.hpp
+    include/cvs/logger/configtypes.hpp
+    include/cvs/logger/tools/fpslogger.hpp
+
+    src/default/defaultfactory.hpp
+
+    src/default/defaultfactory.cpp
+    src/tools/fpslogger.cpp
+    src/configtypes.cpp
+    src/loggerfactory.cpp
+    src/ilogger.cpp)
+
 
 target_link_libraries(${PROJECT_NAME}
-    PUBLIC
-        spdlog::spdlog
-        $<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_core>
-        $<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_imgcodecs>
-        systemd
-    )
+  PUBLIC
+    spdlog::spdlog
+    PkgConfig::systemd
+    $<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_core>
+    $<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_imgcodecs>)
```

Comment:
you can do this in one line, using spaces

Reply:
What's the problem with several lines?

Reply:
More code to maintain, less beautiful, makes readers think why would you write this (reread the line to check if the prefix is copy-pasted or actually differs from the previous line)

Reply:
> More code to maintain

\n instead of a space doesn't make more code

> less beautiful, makes readers think why would you write this (reread the line to check if the prefix is copy-pasted or actually differs from the previous line)

Is it just for you or for everyone?

I think we need a code style for cmake. Without it, this topic has no meaning.

Reply:
@ssysoenko I think @roman-orekhov means you can link the two opencv libraries in one expression without repeating the condition, like this:
```
    $<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_core opencv_imgcodecs>)
```

Reply:
Obviously yes.

Reply:
It doesn't work for me. I revert it. 
`ninja: error: build.ninja:150: bad $-escape (literal $ must be written as $$)
    LINK_LIBRARIES = /usr/lib/x86_64-linux-gnu/libspdlog.so.1.5.0  $<1:"op...
                                                                   ^ near here`

Reply:
@ssysoenko this one works fine with both make and ninja:
```cmake
     "$<$<BOOL:${CVSLOGGER_OPENCV_IMG}>:opencv_imgcodecs;opencv_core>")
```